### PR TITLE
Adding the IIS Web App Management task for machine group

### PR DESCRIPTION
**Pull request by asranja**

### CODE REVIEW

File: Tasks/IISWebAppManagement/task.json [hunk 1]
```diff
@@ -0,0 +1,307 @@
+{
+	"id": "1b2aec60-dc49-11e6-9b76-63056e018cac",
+	"name": "IISWebAppManagement",
+	"friendlyName": "IIS Web App Manage On Machine Group",
+	"description": "Create or update a Website and an App Pool on an IIS using machine group",
+	"author": "Microsoft Corporation",
+	"helpMarkDown": "Replace with markdown to show in help",
```
Comment (asranja) [inline Tasks/IISWebAppManagement/task.json]: Will replace with forward link to readme.

File: Tasks/IISWebAppManagement/IISWebAppManagement.ps1 [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    if ($createAppPool -and $appPoolIdentity -ieq "SpecificUser") {
+        $appPoolPassword = Escape-SpecialChars -str $appPoolPassword
+        $appPoolPassword = "$appPoolPassword" | ConvertTo-SecureString -AsPlainText -Force
+        $appPoolCredentials = New-Object System.Management.Automation.PSCredential ("$appPoolUsername", $appPoolPassword)
+    }
+
+    $appCmdCommands = Escape-SpecialChars -str $appCmdCommands
+
+    Execute-Main -CreateWebsite $createWebsite -WebsiteName $websiteName -WebsitePhysicalPath $websitePhysicalPath -WebsitePhysicalPathAuth $websitePhysicalPathAuth -websitePhysicalPathAuthCredentials $websitePhysicalPathAuthCredentials -AddBinding $addBinding -Protocol $protocol -IpAddress $ipAddress -Port $port -HostName $hostName -ServerNameIndication $serverNameIndication -SslCertThumbPrint $sslCertThumbPrint -CreateAppPool $createAppPool -AppPoolName $appPoolName -DotNetVersion $dotNetVersion -PipeLineMode $pipeLineMode -AppPoolIdentity $appPoolIdentity -AppPoolCredentials $appPoolCredentials -AppCmdCommands $appCmdCommands
```
Comment (arjgupta) [inline Tasks/IISWebAppManagement/IISWebAppManagement.ps1]: Why do we need to have physical path credentials ? Doesn't the way in which we configure the machine group agent ensure that we have access to the path?

File: Tasks/IISWebAppManagement/task.json [hunk 3]
```diff
@@ -0,0 +1,307 @@
+{
+	"id": "1b2aec60-dc49-11e6-9b76-63056e018cac",
```
Comment (arjgupta) [inline Tasks/IISWebAppManagement/task.json]: This should be a new Guid

File: Tasks/IISWebAppManagement/make.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (arjgupta) [inline Tasks/IISWebAppManagement/make.json]: Can you check whether there's a way to bundle appcmd.exe along with the task. If Yes, then should we do it ?

File: Tasks/IISWebAppManagement/task.json (hunk 3 above)
Comment (kmkumaran) [inline Tasks/IISWebAppManagement/task.json]: @arjgupta - Is this guid used in any other task?

File: Tasks/IISWebAppManagement/task.json [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+			"name": "AppCmdCommands",
+			"type": "multiLine",
+			"label": "Additional appCmd.exe commands",
+			"required": false,
+			"groupName": "advanced",
+			"defaultValue": "",
+			"helpMarkDown": "Enter additional AppCmd.exe commands to set the website or the application pool properties. For more than one command use a line separator, like <br/> list apppools <br/> list sites<br/> recycle apppool /apppool.name:ExampleAppPoolName"
+		},
+		{
+			"name": "DeployInParallel",
```
Comment (kmkumaran) [inline Tasks/IISWebAppManagement/task.json]: This option is not relevant in machine group

File: make-options.json [hunk 6]
```diff
@@ -31,6 +31,7 @@
         "Grunt",
         "Gulp",
         "IISWebAppDeploymentOnMachineGroup",
+        "IISWebAppManagement",
```
Comment (kmkumaran) [inline make-options.json]: Rename to IISwebAppManaagementOnMachineGroup

File: Tasks/IISWebAppManagement/IISWebAppManagement.ps1 [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    $deployInParallel = Get-VstsInput -Name "DeployInParallel"
+
+    $hostName = Get-HostName -protocol $protocol -hostNameWithHttp $hostNameWithHttp -hostNameWithSNI $hostNameWithSNI -hostNameWithOutSNI $hostNameWithOutSNI -sni $serverNameIndication
+    
+    Trim-Inputs -siteName ([ref]$websiteName) -physicalPath ([ref]$websitePhysicalPath)  -poolName ([ref]$appPoolName) -websitePathAuthuser ([ref]$websiteAuthUserName) -appPoolUser ([ref]$appPoolUsername) -sslCertThumbPrint ([ref]$sslCertThumbPrint)
+
+    Validate-Inputs -createWebsite $createWebsite -websiteName $websiteName -createAppPool $createAppPool -appPoolName $appPoolName -addBinding $addBinding -protocol $protocol -sslCertThumbPrint $sslCertThumbPrint
+    
+    if ($createWebsite -and $websitePhysicalPathAuth -ieq "WebsiteWindowsAuth") {
+        $websiteAuthUserPassword = Escape-SpecialChars -str $WebsiteAuthUserPassword
```
Comment (kmkumaran) [inline Tasks/IISWebAppManagement/IISWebAppManagement.ps1]: we should be using splatting technique and not worry about escaping arguments.

File: Tasks/IISWebAppManagement/IISWebAppManagement.ps1 [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        $websitePhysicalPathAuthCredentials = New-Object System.Management.Automation.PSCredential ("$websiteAuthUserName", $websitePhysicalPathAuthPassword)
+    }
+
+    if ($createAppPool -and $appPoolIdentity -ieq "SpecificUser") {
+        $appPoolPassword = Escape-SpecialChars -str $appPoolPassword
+        $appPoolPassword = "$appPoolPassword" | ConvertTo-SecureString -AsPlainText -Force
+        $appPoolCredentials = New-Object System.Management.Automation.PSCredential ("$appPoolUsername", $appPoolPassword)
+    }
+
+    $appCmdCommands = Escape-SpecialChars -str $appCmdCommands
```
Comment (kmkumaran) [inline Tasks/IISWebAppManagement/IISWebAppManagement.ps1]: use splatting for passing appcmd commands.

File: Tasks/IISWebAppManagement/IISWebAppManagement.ps1 [hunk 9]
```diff
@@ -0,0 +1,66 @@
+[CmdletBinding()]
+param()
```
Comment (kmkumaran) [inline Tasks/IISWebAppManagement/IISWebAppManagement.ps1]: L0s should be added.

File: Tasks/IISWebAppManagement/task.json (hunk 3 above)
Comment (asranja) [inline Tasks/IISWebAppManagement/task.json]: This is a new guid generated by "tfx build tasks create"

**Pull request merged**